js/index.js

```diff
@@ -37,6 +37,201 @@ const siteContent = {
   },
 };
 
-// Example: Update the img src for the logo
-let logo = document.getElementById("logo-img");
-logo.setAttribute('src', siteContent["nav"]["img-src"])
+
+
+//////////////////////////////////////////////////////////////////////////////////////
+// BTW MAP CAN DO ANYTHING IN THE WORLD - DOESN'T MEAN YOU SHOULD //
+// To be honest, using map is pretty unnecessary with the nav, or elsewhere for that matter. 
+// I first wrote this using forEach, which as it turns out would have worked fine.
+// Originally, I thought I was going to need to modify or append to 
+// each value in the array-like object. Then I read the docs on setAttribute.
+// I realized that we are working with siteContent, which is, more or less, a bunch of strings. 
+// Meaning, in this context, all I needed was some bracket notation and I could append to the
+// output href link. I was funamentally misunderstanding what setAttribute was doing.
+// I also learned that bracket notation within another is okay.
+// All this to add .html to each link, lol!  Good practice though, this experience cleared up a 
+// lot of my syntax hangups, I believe.
+const navLinks = document.querySelectorAll('nav a');  // Select all the 'a' within 'nav'
+const navMadeArr = Array.from(navLinks); // Had problems using map on an array-like thing. This allows me to use map.
+const navArr = navMadeArr.map((element, index) => { // map creates a new array called navArr
+  element.textContent = siteContent['nav'][`nav-item-${index+1}`];
+  element.setAttribute('href', `${siteContent['nav'][`nav-item-${index+1}`]}.html` );
+  return element; // Technically, this works for our purposes without the return. But it just seems like the right thing to do.
+});
+
```

Comment:
All of this is true, but you like to add complexity, so that's OK. :P
I think you could have probably also used .toString() on each element, and then use the strings. But we already had the strings available in the object literal, which is why forEach is acceptable here. I think the other thing is that Josh was wanting us to realize that querySelectorAll returns a NodeList, which is an array-like object, and NodeList has the forEach method on it's class constructor.